code/modules/mob/living/carbon/carbon.dm

```diff
@@ -939,8 +939,8 @@
 	if(heal_flags & HEAL_LIMBS)
 		regenerate_limbs()
 
-	if(heal_flags & (HEAL_REFRESH_ORGANS|HEAL_ORGANS))
-		regenerate_organs(regenerate_existing = (heal_flags & HEAL_REFRESH_ORGANS))
+	if(heal_flags & (HEAL_REFRESH_ORGANS|HEAL_REFRESH_ORGANS|HEAL_ORGANS))
+		regenerate_organs(regenerate_existing = !!(heal_flags & HEAL_REPLACE_ORGANS), remove_hazardous = !!(heal_flags & HEAL_REFRESH_ORGANS))
```

Comment:
& doesn't return boolean according to my testing

Reply:
& is binary AND, so it returns a number but should still work

Reply:
Truthy sure, but you expect a boolean in the arguments, not a number

Reply:
It doesn't matter